packages/edit-site/src/components/header/index.js

```diff
@@ -1,24 +1,66 @@
 /**
  * WordPress dependencies
  */
+import { useCallback } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { BlockNavigationDropdown, ToolSelector } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
+import { useEditorContext } from '../editor';
+import TemplateSwitcher from '../template-switcher';
 import SaveButton from '../save-button';
 
 export default function Header() {
+	const { settings, setSettings } = useEditorContext();
+	const setActiveTemplateId = useCallback(
+		( newTemplateId ) =>
+			setSettings( ( prevSettings ) => ( {
+				...prevSettings,
+				templateId: newTemplateId,
+				templateType: 'wp_template',
+			} ) ),
+		[]
+	);
+	const setActiveTemplatePartId = useCallback(
+		( newTemplatePartId ) =>
+			setSettings( ( prevSettings ) => ( {
+				...prevSettings,
+				templateId: newTemplatePartId,
+				templateType: 'wp_template_part',
+			} ) ),
+		[]
+	);
+	const addTemplateId = useCallback(
+		( newTemplateId ) =>
+			setSettings( ( prevSettings ) => ( {
+				...prevSettings,
+				templateId: newTemplateId,
+				templateIds: [ ...prevSettings.templateIds, newTemplateId ],
+			} ) ),
+		[]
+	);
 	return (
 		<div
 			className="edit-site-header"
 			role="region"
 			aria-label={ __( 'Site editor top bar.' ) }
 			tabIndex="-1"
 		>
-			<h1 className="edit-site-header__title">
-				{ __( 'Site Editor' ) } { __( '(beta)' ) }
-			</h1>
+			<div className="edit-site-header__toolbar">
+				<TemplateSwitcher
+					ids={ settings.templateIds }
+					templatePartIds={ settings.templatePartIds }
+					activeId={ settings.templateId }
+					isTemplatePart={ settings.templateType === 'wp_template_part' }
+					onActiveIdChange={ setActiveTemplateId }
+					onActiveTemplatePartIdChange={ setActiveTemplatePartId }
+					onAddTemplateId={ addTemplateId }
```

Comment:
Seems to me like most of these props are useless, why not just do the switching in the component itself since you have access to the Context (or store if it's refactored)

Reply:
Because I didn't want to prematurely tie these components to a specific context when other UIs could use them in the future.

Reply:
It's a bit different pattern from the other components render right after it. I like the fact that the header becomes more about composing things together and keepnig the logic where it belongs (template switching in TemplateSwitcher component). 

I don't really see us reusing that component differently in other places.